### PR TITLE
Bugfix: Respect PATH env variable in zypper modules

### PR DESCRIPTION
**Pull request by sealor**
##### SUMMARY
Currently the zypper modules use the hard-coded path `/usr/bin/zypper` for the zypper executable. So, in case of other locations the modules are not able to handle zypper actions correctly.

This Pull Request fixes this issue. Now, the zypper modules respect the PATH environment variable to find the zypper exceutable.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- community.general.zypper
- community.general.zypper_repository

### CODE REVIEW

File: plugins/modules/packaging/os/zypper.py [hunk 1]
```diff
@@ -336,7 +337,7 @@ def get_cmd(m, subcommand):
     "puts together the basic zypper command arguments with those passed to the module"
     is_install = subcommand in ['install', 'update', 'patch', 'dist-upgrade']
     is_refresh = subcommand == 'refresh'
-    cmd = ['/usr/bin/zypper', '--quiet', '--non-interactive', '--xmlout']
+    cmd = [find_executable('zypper'), '--quiet', '--non-interactive', '--xmlout']
```
Comment (felixfontein) [inline plugins/modules/packaging/os/zypper.py:340]: Please use `m.get_bin_path` instead (https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1406). Same for `zypper_repository` (you'll need to pass the `AnsibleModule` instance to `_get_cmd()` there though).

You also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

Comment (sealor) [reply plugins/modules/packaging/os/zypper.py]: Thank you for your review! I improved the PR according to your recommendations.

File: changelogs/fragments/2094-bugfix-respect-PATH-env-variable-in-zypper-modules.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - zypper_repository - respect PATH environment variable when resolving zypper executable path.
+  - zypper - respect PATH environment variable when resolving zypper executable path.
```
Comment (felixfontein) [inline changelogs/fragments/2094-bugfix-respect-PATH-env-variable-in-zypper-modules.yaml:3]:
```suggestion
  - zypper, zypper_repository - respect ``PATH`` environment variable when resolving zypper executable path (https://github.com/ansible-collections/community.general/pull/2094).
```

**Comment by russoz** [comment]
LGTM

**Review by evrardjp — COMMENTED** [review]
I didn't check in details, but it doesn't sound bad to me.

**Comment by felixfontein** [comment]
@sealor thanks for fixing this!
@russoz @evrardjp thanks for reviewing!

**Comment by Andersson007** [comment]
@sealor hi, thanks for your contribution! We could add your GH login to `.github/BOTMETA.yml` in a corresponding team as a maintainer.
You'll be notified about Issues/PRs related to this stuff and your `shipit` will be counted by bot for automerge (needs two for bugfixes and minor changes).
What do you think?

**Comment by sealor** [comment]
> @sealor hi, thanks for your contribution! We could add your GH login to `.github/BOTMETA.yml` in a corresponding team as a maintainer.
> You'll be notified about Issues/PRs related to this stuff and your `shipit` will be counted by bot for automerge (needs two for bugfixes and minor changes).
> What do you think?

@Andersson007 
Thank you for your invitation. Yes, I would be glad to support the further development!

**Comment by Andersson007** [comment]
@sealor great, thanks! Could you please approve the PR https://github.com/ansible-collections/community.general/pull/2354?

**Pull request merged**